### DISCUSSION TITLE
[Review Request] How to prepend protocol to URLs created in the Link Editor?

### DISCUSSION THREAD

**Discussion by elatomo** [mdx-editor/editor#614] [Q&A]
Hi! First of all, thank you for such a great library :heart:.

I would like to extend the link editor to ensure that external links created through it, which do not have a protocol, are treated as external links. Essentially, I want to prepend `https://` to all URLs created in the link editor.

I believe this should be resolved by defining a custom plugin and using both the `linkDialogState$` cell and the `updateLink$` signal.

So far, this is the only working solution I've come up with, which relies on `addComposerChild$` to add a React component that uses the `linkDialogState$` cell and the `updateLink$` signal to update the URLs, and honestly, it feels completely overcomplicated.

Do you have any advice on how to achieve this in a cleaner, more elegant way?

Thanks a lot in advance!

```typescript
export const ensureExternalLinkPlugin = realmPlugin({
  init(realm) {
    realm.pubIn({
      [addComposerChild$]: () => {
        const link = useCellValue(linkDialogState$);
        const updateLink = usePublisher(updateLink$);

        useEffect(() => {
          const urlIsInternal =
            link.type === "preview" &&
            link.url !== "" &&
            !link.url.startsWith("http");
          if (urlIsInternal) {
            updateLink({
              title: link.title,
              url: `https://${link.url}`,
            });
          }
        }, [link]);

        return null;
      },
    });
  },
});

**Comment by yaffalhakim1** [comment]
hello, do you find the answer?

**Reply by elatomo** [reply]
@yaffalhakim1, I wasn't able to find a better solution than the one I mentioned in my question.
